contrib/opencensus-ext-azure/opencensus/ext/azure/common/__init__.py

```diff
@@ -39,3 +40,21 @@ class Options(BaseObject):
         storage_retention_period=7*24*60*60,
         timeout=10.0,  # networking timeout in seconds
     )
+
+
+def validate_key(instrumentation_key):
+    if not instrumentation_key:
+        raise ValueError("Instrumentation key cannot be none or empty.")
+    # Validate UUID format
+    # Specs taken from https://tools.ietf.org/html/rfc4122
+    pattern = re.compile('[0-9a-f]{8}-'
+                         '[0-9a-f]{4}-'
+                         '[1-5][0-9a-f]{3}-'
+                         '[89ab][0-9a-f]{3}-'
+                         '[0-9a-f]{12}')
+    # re.fullmatch not available for python2
+    # We use re.match, we matches from the beginning, and then do a length
+    # check to ignore everything afterward.
+    match = pattern.match(instrumentation_key)
```

Comment:
This is weird, we should use something `^foobar$`.

Reply:
Sorry, can you explain a bit more about what you mean?

Reply:
^ Start of string, or start of line in multi-line pattern
$ End of string, or end of line in multi-line pattern

Reply:
I've add the `^` and `$` to the regex so length does not need to be checked.